webapp/src/main/java/com/box/l10n/mojito/service/ai/translation/AITranslateCronJob.java

```diff
@@ -295,7 +296,7 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
                                   logger.debug(
                                       "Sending pending MT for tmTextUnitId: {} for deletion",
                                       pendingMT.getTmTextUnitId());
-                                  aiTranslationService.sendForDeletion(pendingMT);
+                                  processedMTs.add(pendingMT);
```

Comment:
processedMTs will be accessed from different threads - we should look at making sure the add is thread safe given that the access patterns

Nit: re naming - instead of processedMTs - could we use something like textUnitsToClearPendingMT ? (even we add them to the list - they're still yet to be processed)

Reply:
Good catch on the list addition!